### PR TITLE
kristall: fix build on darwin

### PR DESCRIPTION
**Pull request by sikmir**
###### Motivation for this change
ZHF: #144627 (@NixOS/nixos-release-managers)

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sikmir** [comment]
@veprbl What is wrong with `libsForQt5.mkDerivation`?

**Comment by risicle** [comment]
Doesn't build for me on nixos x86_64 now:

```
builder for '/nix/store/gcqv40a037lm05a8jqv62yjizm2cp6ka-kristall-0.3.drv' failed with exit code 1; last 3 log lines:
  qmakePrePhase
  qtPreHook
  Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.
```

File: pkgs/applications/networking/browsers/kristall/default.nix [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
+    rev = "V${version}";
     sha256 = "07nf7w6ilzs5g6isnvsmhh4qa1zsprgjyf0zy7rhpx4ikkj8c8zq";
   };
 
-  buildInputs = [ qtbase qtmultimedia ];
+  postPatch = lib.optionalString stdenv.cc.isClang ''
+    sed -i '1i #include <errno.h>' src/browsertab.cpp
+  '';
+
+  nativeBuildInputs = [ qmake ];
```
Comment (veprbl) [inline pkgs/applications/networking/browsers/kristall/default.nix:18]:
```suggestion
  nativeBuildInputs = [ wrapQtAppsHook qmake ];
```

**Comment by veprbl** [comment]
> @veprbl What is wrong with `libsForQt5.mkDerivation`?

It's not composable, so I'd love to see it deprecated.

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 146531` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kristall</li>
  </ul>
</details>

**Review by veprbl — APPROVED** [review]

File: pkgs/applications/networking/browsers/kristall/default.nix [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
     repo = "kristall";
-    rev = "V" + version;
+    rev = "V${version}";
     sha256 = "07nf7w6ilzs5g6isnvsmhh4qa1zsprgjyf0zy7rhpx4ikkj8c8zq";
   };
 
-  buildInputs = [ qtbase qtmultimedia ];
+  postPatch = lib.optionalString stdenv.cc.isClang ''
+    sed -i '1i #include <errno.h>' src/browsertab.cpp
+  '';
```
Comment (jonringer) [inline pkgs/applications/networking/browsers/kristall/default.nix:16]: not a blocker for this PR, but we should probably create an issue upstream, and then they can use some directives to optionally include this.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 146531` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kristall</li>
  </ul>
</details>

File: pkgs/applications/networking/browsers/kristall/default.nix (hunk 2 above)
Comment (sikmir) [reply pkgs/applications/networking/browsers/kristall/default.nix:16]: There are ~400 commits after V0.3, it maybe already fixed...

**Pull request merged**